ScoutSuite/providers/azure/provider.py

```diff
@@ -37,37 +38,16 @@ def __init__(self, project_id=None, organization_id=None,
 
     def authenticate(self, key_file=None, user_account=None, service_account=None, **kargs):
         """
-        Implement authentication for the GCP provider
-        Refer to https://google-auth.readthedocs.io/en/stable/reference/google.auth.html.
+        Implements authentication for the Azure provider using azure-cli.
+        Refer to https://docs.microsoft.com/en-us/python/azure/python-sdk-azure-authenticate?view=azure-python.
 
         :return:
         """
 
         try:
-
-            # TODO this is temporary
-            import os
-            # Azure subscription
-            SUBSCRIPTION_ID = os.environ['SUBSCRIPTION_ID']
-            # Tenant ID for your Azure Subscription
-            TENANT_ID = os.environ['TENANT_ID']
-            # Your Service Principal App ID
-            CLIENT = os.environ['CLIENT']
-            # Your Service Principal Password
-            KEY = os.environ['KEY']
-
-            self.aws_account_id = TENANT_ID # TODO this is for AWS
-
-            credentials = ServicePrincipalCredentials(
-                client_id=CLIENT,
-                secret=KEY,
-                tenant=TENANT_ID
-            )
-
-            self.credentials = AzureCredentials(credentials, SUBSCRIPTION_ID)
-
+            cli_credentials, self.aws_account_id = get_azure_cli_credentials()          #TODO: Remove aws_account_id
```

Comment:
Can you complete that todo in this PR or is it out of the scope?

Reply:
As with the other issue, this one is used in __main__.py at line 132 and might lead to a refactoring rabbit hole. Again, I believe it would be better for it to have it's own PR and keep this one only for adding the new feature.

Reply:
Same here 👍